lisk-dex-electron/src/PlaceOrder.jsx

```diff
@@ -29,10 +29,12 @@ export default class PlaceOrder extends React.Component {
     const { target } = event;
     const value = target.type === 'checkbox' ? target.checked : target.value;
     const { name } = target;
-
-    this.setState({
-      [name]: value,
-    });
+    const isNumber = value === '' || /^-?[\d.]+(?:e-?\d+)?$/.test(value);
```

Comment:
This RegExp has some problems. It will say that `123.234.34` (with two dots) is a number. Also, we should not allow negative numbers in this case.

On the DEX back end, this RegExp is used:

`/^([0-9]+\.?|[0-9]*\.[0-9]+)$/`

It allows a number with or without decimal. It allows ending the number with a dot like `10.` or starting with a dot like `.2` but JavaScript also allows both of these cases.

Reply:
You could also use `/^([\d]+\.?|[\d]*\.[\d]+)$/`